lib/usecase/get_survey_use_case.dart

```diff
@@ -25,13 +25,16 @@ class GetSurveysUseCase extends UseCase<List<SurveyModel>, GetSurveysInput> {
   @override
   Future<Result<List<SurveyModel>>> call(GetSurveysInput params) {
     return _repository
-        .getSurveys(
-          number: params.pageNumber,
-          size: params.pageSize,
-        )
-        // ignore: unnecessary_cast
-        .then((value) => Success(value) as Result<List<SurveyModel>>)
-        .onError<NetworkExceptions>(
+        .getSurveys(number: params.pageNumber, size: params.pageSize)
+        .then((value) {
+      _saveSurveys(value);
+      // ignore: unnecessary_cast
+      return Success(value) as Result<List<SurveyModel>>;
+    }).onError<NetworkExceptions>(
```

Comment:
Since you changed the purpose of this use case, should we rename it? `get_and_cach_survey_use_case`

Another approach is moving the saving surveys to repository scope and keep it private

Reply:
Good catch @hoangnguyen92dn I agreed 🚀